visualization/tools/pcd_viewer.cpp

```diff
@@ -422,6 +428,19 @@ main (int argc, char** argv)
     if (pcd.read (argv[p_file_indices.at (i)], *cloud, origin, orientation, version) < 0)
       return (-1);
 
+    // Calculate transform if available.
+    if (pose_x.size () > i && pose_y.size () > i && pose_z.size () > i &&
+        pose_roll.size () > i && pose_pitch.size () > i && pose_yaw.size () > i)
+    {
```

Comment:
Good idea to allow just translation. I would then just check the translation and rotation vector sizes separately, and assume an identity rotation matrix where none is given. I've just pushed a commit to my clone doing that, how do you like it?